amivapi/tests/events/test_emails.py

```diff
@@ -56,7 +56,9 @@ def test_email_signup_delete(self):
 
         # With redirect set
         self.app.config['SIGNUP_DELETED_REDIRECT'] = "somewhere"
-        self.api.get('/delete_signup/%s' % token, status_code=302)
+        self.api.get('/eventsignups/%s' % signup['_id'], status_code=200)
+        self.api.get('/delete_signup/%s' % token, status_code=200)
+        self.api.post('/delete_confirmed/%s' % token, status_code=302)
```

Comment:
Thank you for adding tests ... unfortunately I do not understand at all what is going on here, and why e.g. in this last line a status code `302` is expected.

As the flow is now considerably more complicated, please either:

- Split this test up into multiple tests that are clear about which part is being tested.
- If it is impossible to split them up, please modify the docstring to explain the expected behavior.

The more I look at it, the more I think our original tests here are already confusing. I would greatly appreciate it if you could try to split the tests up a bit more to make them more understandable.